core/domain/suggestion_services.py

```diff
@@ -2959,3 +2960,203 @@ def enqueue_contributor_ranking_notification_email_task(
+        from_date: datetime.datetime. The start of the date range for which
+            the contributions were created.
+        to_date: datetime.datetime. The end of the date range for which
+            the contributions were created.
+        user_id: str. The user ID of the contributor.
+
+    Returns:
+        Dict[str, Union[str, None]]. Data to generate translation submitter
+        certificate.
+
+    Raises:
+        Exception. The language is invalid.
+    """
+    signature = feconf.TRANSLATION_TEAM_LEAD
+
+    # Adds one date to the to_date to make sure the contributions within
+    # the to_date are also counted for the certificate.
+    to_date_to_fetch_contributions = to_date + datetime.timedelta(days=1)
+
+    language = next(filter(
+        lambda lang: lang['id'] == language_code,
+        constants.SUPPORTED_AUDIO_LANGUAGES), None)
+    if language is None:
+        raise Exception('The provided language is invalid.')
+    # Since we have found a corresponding language for the given language code
+    # we can make sure that language code is not None.
+    assert isinstance(language_code, str)
+    language_description = language['description']
+    if language_description.find(' (') != -1:
+        language_description = language_description[
+            language_description.find('(') + 1:language_description.find(')')]
+
+    suggestions = (
+        suggestion_models.GeneralSuggestionModel
+        .get_translation_suggestions_submitted_within_given_dates(
+            from_date,
+            to_date_to_fetch_contributions,
+            user_id,
+            language_code
+        )
+    )
+
+    words_count = 0
+    for model in suggestions:
+        suggestion = get_suggestion_from_model(model)
+
+        # Retrieve the html content that is emphasized on the
+        # Contributor Dashboard pages. This content is what stands
+        # out for each suggestion when a user views a list of
+        # suggestions.
+        get_html_representing_suggestion = (
+            SUGGESTION_EMPHASIZED_TEXT_GETTER_FUNCTIONS[
+                suggestion.suggestion_type]
+        )
+        plain_text = _get_plain_text_from_html_content_string(
+            get_html_representing_suggestion(suggestion))
+
+        words = plain_text.split(' ')
+        words_without_empty_strings = [
+            word for word in words if word != '']
+        words_count += len(words_without_empty_strings)
+    # Source: https://github.com/oppia/oppia/pull/16513/files#r1027109026
+    hours_contributed = round((words_count / 300), 2)
+
+    if words_count == 0:
+        raise Exception(
+            'There are no contributions for the given time range.')
+
+    return {
+        'from': from_date.strftime('%d %b %Y'),
+        'to': to_date.strftime('%d %b %Y'),
+        'contribution_hours': str(hours_contributed),
+        'team_lead': signature,
+        'language': language_description
+    }
+
+
+def _generate_question_contributor_certificate(
```

Comment:
I think you should change this function's name to something like "generate certificate data" and define a domain object for the return value that gets changed using to_dict() in the controller to pass to the frontend.

That's because what's being returned here is not a certificate, but the data for a certificate. Make sure that the whole thing is conceptually correct.

Reply:
Done

![certificate (28)](https://user-images.githubusercontent.com/41409951/206944500-d1b48276-a139-49bd-b2f6-7771d3665ddf.jpeg)
![certificate (29)](https://user-images.githubusercontent.com/41409951/206944515-ee6bcb59-473f-46b0-9f8b-2baf36c5efbd.jpeg)